test/e2e/antreapolicy_test.go

```diff
@@ -1945,6 +1945,9 @@ func testANPMultipleAppliedTo(t *testing.T, singleRule bool) {
 
 	t.Logf("Making the Policy apply to y/c by labeling it with the temporary label that matches the dummy AppliedTo")
 	podYC, _ := k8sUtils.GetPodByLabel("y", "c")
+	if podYC == nil {
```

Comment:
I think the more "Go" way is to receive the error (instead of ignoring it with `_`), check the error and appened it in the below error log.
But I'm more curious why it would fail to get the Pod. Doesn't it check the test pods in the first place after creating them?

Reply:
in my test, the test suit creates a pod first, and delete them soon, and later when the test suit wants to get the pod, it gets nil pod.

The reason I judge the pod is because the function is implemented as:
```
func (k *KubernetesUtils) GetPodByLabel(ns string, name string) (*v1.Pod, error) {
	pods, err := k.getPodsUncached(ns, "pod", name)
	if err != nil {
		return nil, errors.WithMessagef(err, "unable to get Pod in Namespace %s with label pod=%s", ns, name)
	}
	if len(pods) == 0 {
		return nil, nil
	}
	return &pods[0], nil
}
```

it can return nil pod without error, @tnqn do you mean to return error when len(pods) is 0?

Reply:
> in my test, the test suit creates a pod first, and delete them soon, and later when the test suit wants to get the pod, it gets nil pod.

Is it expected to delete the pods then use them later? @Dyanngg 

> it can return nil pod without error, @tnqn do you mean to return error when len(pods) is 0?

I think so, it can make client simpler.